lib/util/git/index.ts

```diff
@@ -626,18 +629,23 @@ export async function isBranchModified(branchName: string): Promise<boolean> {
   logger.debug('branch.isModified(): using git to calculate');
 
   await syncGit();
-  // Retrieve the author of the most recent commit
-  let lastAuthor: string | undefined;
+  // Retrieve the commit authors
+  let branchAuthors: string[] = [];
   try {
-    lastAuthor = (
-      await git.raw([
-        'log',
-        '-1',
-        '--pretty=format:%ae',
-        `origin/${branchName}`,
-        '--',
-      ])
-    ).trim();
+    branchAuthors = [
+      ...new Set(
+        (
+          await git.raw([
+            'log',
+            '--pretty=format:%ae',
+            `origin/${branchName}...origin/${baseBranch}`,
```

Comment:
We've been using `origin/` since the times where we did shallow commits and couldn't sure that commits were visible to us. But I think now with a blobless clone, we maybe don't need `origin/`. Not sure if now is the time to change though

Reply:
Yeah, that's fair. Happy to update it to be `${branchName}...${baseBranch}`, which could introduce some risk; although it appears to work for me locally

Reply:
could be an issue if repos are retrained locally and then the remote branch was changed.

https://docs.renovatebot.com/self-hosted-configuration/#persistrepodata

Reply:
Retrained? Rebased?

I think if users persist data and change it then they are in for pain..

Reply:
repos are persisted locally between runs and the user changes a remote branch, not a local.
that way the local and remote branch can be conflicted.
I'm not sure how we handle that case today.

the current diff with origin will compare the remote branches, but without it would compare the local branches if exists.

Reply:
ok

Reply:
Let's leave as origin